src/frontend/utils/conversejs/converse-plugins/logoutPlugin.ts

```diff
@@ -0,0 +1,20 @@
+import { converse } from './../../window';
+
+const PLUGIN_NAME = 'logout-plugin';
+
+const addLogoutPlugin = () =>
+  converse.plugins.add(PLUGIN_NAME, {
+    dependencies: ['converse-muc'],
+    initialize() {
+      const _converse = this._converse;
+
+      window.addEventListener('beforeunload', () => {
+        _converse.api.user.logout();
```

Comment:
That was a hack I made to not loose a session. We should look if it's still needed with what you are doing. For now we can keep it but keep in mind to test without this plugin once you work is ended

Reply:
Yep, I think i will also see this comment in my second PR, because the full chat is available and displayed, so I think it be easier to perform changes.